pages/cart.js

```diff
@@ -0,0 +1,232 @@
+import { useContext, useEffect, useState } from "react";
+import styled from "styled-components";
+import { toast } from "react-toastify";
+import { FaTimes } from "react-icons/fa";
+
+import Header from "../components/Header";
+import Footer from "../components/Footer";
+import { Divider, Button } from "../styles/UIKit";
+import useApi from "../utils/useApi";
+import CartConstants from "../config/CartConfig";
+import AuthContext from "../contexts/AuthContext";
+
+const Cart = () => {
+  const api = useApi();
+
+  const [cartData, setCartData] = useState({});
+  const { cart, updateCart } = useContext(AuthContext);
+
+  const fetchData = () => {
+    try {
+      setCartData(cart);
+    } catch (err) {
+      toast.error("Cannot fetch cart data");
+      console.log("Error while fetching cart data : ", err);
+    }
+  };
+
+  const handleDeleteCartItem = (itemId) => {
+    deleteCartItem(itemId);
+  };
+
+  const deleteCartItem = async (itemId) => {
+    try {
+      const {
+        data: { cart },
+      } = await api.delete(`cart/product/${itemId}`);
+
+      setCartData(cart);
+      updateCart(cart);
+      toast.success("Deleted cart item successfully");
+    } catch (err) {
+      toast.error("Some error has occurred");
+      console.log("Error while deleting cart item : ", err);
+    }
+  };
+
+  useEffect(() => {
+    fetchData();
+  }, []);
+
+  return (
+    <CartWrapper>
+      <Header />
+      <CartContainer className={cartData?.items?.length === 0 && "hide"}>
+        <CartTitleContainer>
+          <FlexAdjusterCloseX>
+            <EmptyBox />
+          </FlexAdjusterCloseX>
+          <FlexAdjuster>
+            <EmptyBox />
+          </FlexAdjuster>
+          <FlexAdjuster>
+            {" "}
+            <CartHeading> {CartConstants.header.product} </CartHeading>
+          </FlexAdjuster>
+          <FlexAdjuster>
+            <CartHeading> {CartConstants.header.price} </CartHeading>
+          </FlexAdjuster>
+          <FlexAdjuster>
+            <CartHeading> {CartConstants.header.quantity} </CartHeading>
+          </FlexAdjuster>
+          <FlexAdjuster>
+            <CartHeading> {CartConstants.header.total} </CartHeading>
+          </FlexAdjuster>
+        </CartTitleContainer>
+        <CartDivider />
+        {cartData?.items?.map((cartItem) => (
+          <div key={cartItem?._id}>
+            <CartContent>
+              <FlexAdjusterCloseX>
+                <CloseX
+                  onClick={() => handleDeleteCartItem(cartItem?.product._id)}
```

Comment:
call deleteCartItem directly, no need to create a new function, that doesn't do anything

Reply:
taken care